Document how to set default for containerConcurrency (#2682)
knative/docs: docs/serving/autoscaling/concurrency.md

### Hard limit

The hard limit is always specified [per revision](./autoscaling-concepts.md) via the `containerConcurrency` field on the revision spec. This particular per-revision setting is not an annotation; it is present on the revision's spec itself as `containerConcurrency`.

There is no global setting for the hard limit in the autoscaling config map, but a default value can be set for the Revision's `containerConcurrency` field in `config-defaults.yaml`.

* The default value is `0`, meaning that there is no limit on the number of requests that are allowed to flow into the revision.
* A value greater than `0` specifies the exact number of requests that are allowed to flow to the replica at any one time.

> * **Global key:** No global key (but a default can be set for the per-revision spec key in `config-defaults.yaml`).

Reviewer: Hmm, let's document the global key in the other configmap then and add an example. We can just add a remark that it's in a different CM.

You might want to add that it's not in the AS configMap because it touches more aspects (namely queuing and buffering which are somewhat concerns of networking) which is also why this isn't just yet another autoscaling annotation but an integral part of the spec IIRC

Author: okie, gave that a go

### Hard limit

> The hard limit is always specified [per revision](./autoscaling-concepts.md) via the `containerConcurrency` field on the revision spec. This particular per-revision setting is not an annotation; it is present on the revision's spec itself as `containerConcurrency`.

Reviewer 2: ```suggestion
The hard limit is specified [per Revision](./autoscaling-concepts.md) using the `containerConcurrency` field on the Revision spec. This setting is not an annotation.
```

> There is no global setting for the hard limit in the autoscaling config map (because `containerConcurrency` has implications outside of autoscaling, such as on buffering and queuing of requests), but a default value can be set for the Revision's `containerConcurrency` field in `config-defaults.yaml`.

Reviewer 2: ```suggestion
There is no global setting for the hard limit in the autoscaling ConfigMap, because `containerConcurrency` has implications outside of autoscaling, such as on buffering and queuing of requests. However, a default value can be set for the Revision's `containerConcurrency` field in `config-defaults.yaml`.
```

    spec:
      containerConcurrency: 50
```
{{< /tab >}}

> {{% tab name="Global (Defaults ConfigMap)" %}}

Reviewer: Operator example too? :pray:

Author: heh, never actually used the operator ☺️, but took an educated guess
